### PR TITLE
[Inductor] Extend Pattern Matcher to Match Equivalent Function Invocation

### LINKED ISSUE #104391: Extend Pattern Matcher to Match Equivalent Function Invocations
### 🐛 Describe the bug

If you add a print statement to `should_pad_addmm` in `pad_mm`.py :

```
--- a/torch/_inductor/fx_passes/pad_mm.py
+++ b/torch/_inductor/fx_passes/pad_mm.py
@@ -82,6 +82,7 @@ def addmm_pattern(input, mat1, mat2, beta, alpha):
 
 
 def should_pad_addmm(match):
+    print("TESTING FOR REPLACEMENT")
     mat1, mat2, input = fetch_fake_tensors(match, ("mat1", "mat2", "input"))
     return should_pad_common(mat1, mat2, input) and should_pad_bench(
         mat1, mat2, torch.ops.aten.addmm, input=input 
```

and then run the following script:

```
import torch

@torch.compile()
def test_mm_padding(inp, x, y):
    return torch.ops.aten.addmm(inp, x, y)

@torch.compile()
def test_mm_padding2(inp, x, y):
    return torch.ops.aten.addmm(inp, x, y, alpha=1.0, beta=1.0)

inps = [torch.rand([20]), torch.rand([20, 20]), torch.rand([20, 20])]
inps = [x.to(device="cuda") for x in inps]

print("Running first")
test_mm_padding(*inps)

print("Running second")
test_mm_padding2(*inps)
```

You'll see that only second addmm invocation which explicitly passes in alpha and beta get matched, even though both addmm invocations are equivalent. That is because we are matching on exact [# of args and kwargs](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/pattern_matcher.py#L343-L347). 

We should extend the pattern matcher so that equivalent invocations get matched. A relevant function here is [normalize_function](https://github.com/pytorch/pytorch/blob/732067e5c30519df5c8906913825bd905e9185fa/torch/fx/operator_schemas.py#L247). 

A couple things you may run into:

- there's an annoying thing where pytorch replaces "self" with "input
- this might break existing replacement patterns that are directly introspecting the fx node. replacement patterns should mostly be fine but graph patterns are more likely to break. we could have an option to the patterns that would canonicalize the calling convention to the convention that its been specified as. There aren't tooooo many existing patterns so it shouldn't be terrible to update them.

shout out @aakhundov for spotting this !

### Versions

master

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #104391

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Comment by yanboliang** [comment]
abandon this in favor of #107832
